app/models/concerns/host_params.rb

```diff
@@ -54,6 +54,7 @@ def host_inherited_params_objects
       params += domain.domain_parameters if domain
       params += operatingsystem.os_parameters if operatingsystem
       params += extract_params_from_object_ancestors(hostgroup) if hostgroup
+      params += subnet.subnet_parameters if subnet
```

Comment:
I don't think subnet parameters should get precedence over  hostgroup parameters so this line should move up. Maybe right after domain parameters? or even before domain?